File: android/androidCourse/Seccion_05_Listados/02_ListView_Personalizado/app/src/main/java/com/miguelcr/a02_listview_personalizado/MainActivity.java
package com.miguelcr.a02_listview_personalizado;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    ListView lista;
    List<Averia> averiaList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        lista = (ListView) findViewById(R.id.listViewAverias);

        averiaList = new ArrayList<>();
        averiaList.add(new Averia("Espejo roto","Audi - A4","http://blog.mister-auto.es/wp-content/uploads/2014/09/23116650_m.jpg",2));
        averiaList.add(new Averia("Paragolpes delantero","Citroen - C4","",0));
        averiaList.add(new Averia("Embrague","Seat - Ibiza","",0));
        averiaList.add(new Averia("Cambio de aceite","Seat - Toledo","",1));

        MiAdaptadorAverias adaptadorAverias = new MiAdaptadorAverias(
                this,
                R.layout.averia_item,
                averiaList
        );

        lista.setAdapter(adaptadorAverias);
        lista.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
                Toast.makeText(MainActivity.this, "valor "+averiaList.get(i).getTitulo()+"", Toast.LENGTH_SHORT).show();
            }
        });

    }


}

File: kotlin/OpenWebinars/Seccion4AndroidMeetsKotlin/Leccion6RecyclerViewYAdapter/app/src/main/java/com/openwebinars/recyclerviewandadapterkotlin/MainActivity.kt
package com.openwebinars.recyclerviewandadapterkotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.GridLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val carList = mutableListOf<CarModel>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initCarList()
        setupRecyclerView()
    }

    private fun setupRecyclerView() {
        mainList.layoutManager = GridLayoutManager(this, 2)
        mainList.adapter = CarAdapter(carList, {
            toast(String.format("Marca %s", it.id), 500)
        })
    }

    private fun initCarList() {
        for (number in 1 until 100) {
            val id = if (number % 2 == 0) number.toString() else ""
            carList.add(CarModel(id, String.format("Marca %s", number)))
        }
    }
}
